### ISSUE TITLE
Bug Report: Regression predictions not denormalized back to original scale

### ISSUE THREAD

**Issue by andreasgoethals** [qile2000/TabBench#89]
## Summary
All classical regression methods in TALENT return predictions in **normalized space** (mean≈0, std≈1) instead of denormalizing them back to the original target scale. This causes severe errors for downstream tasks that expect predictions in the original scale, especially for bounded regression tasks like Loss Given Default (LGD) prediction where targets are in [0, 1].

## Impact
- **Severity**: High - Affects all classical regression methods
- **Affected methods**: LinearRegression, KNN, RandomForest, XGBoost, LightGBM, CatBoost, SVM
- **Data loss**: Predictions are unusable for applications requiring original scale
- **Silent failure**: Metrics (MAE, RMSE) are correctly scaled back, hiding the bug from users

### Step 1: Normalization (TALENT/model/lib/data.py)
```python
def data_label_process(y_data, is_regression, info=None, encoder=None):
    if is_regression:
        mean, std = y_data['train'].mean(), y_data['train'].std()
        y = {k: (v - mean) / std for k, v in y.items()}  # ✓ Normalize
        info = {'policy': 'mean_std', 'mean': mean, 'std': std}
        return y, info, None
```

### Step 2: Training happens on normalized targets
```python
self.model.fit(self.N['train'], self.y['train'])  # y['train'] normalized
```

### Step 3: Predictions returned WITHOUT denormalization
```python
# Example: TALENT/model/classical_methods/lr.py
def predict(self, data, info, model_name):
    test_logit = self.model.predict(self.N_test)  # Normalized space!
    vres, metric_name = self.metric(test_logit, test_label, self.y_info)
    return vres, metric_name, test_logit  # ⚠️ NEVER denormalized
```

### Step 4: Metrics partially scaled back (inconsistent)
```python
# In classical_methods.metric()
if y_info['policy'] == 'mean_std':
    mae *= y_info['std']   # ✓ Scale MAE back
    rmse *= y_info['std']  # ✓ Scale RMSE back
    # ⚠️ BUT predictions themselves are NOT scaled back!
```

## Minimal Reproducible Example
```python
import numpy as np
from sklearn.datasets import make_regression
from TALENT.model.classical_methods.lr import LinearRegressionMethod

# Create synthetic data with target mean=50, std=20
X, y = make_regression(n_samples=1000, n_features=10, noise=10, random_state=42)
y = y * 20 + 50  # Scale to mean≈50

# Split data
train_data = {'train': y[:700], 'val': y[700:900], 'test': y[900:]}

# TALENT normalizes internally
mean, std = train_data['train'].mean(), train_data['train'].std()
print(f"Original: mean={mean:.2f}, std={std:.2f}")

# Train model (gets normalized targets internally)
model = LinearRegressionMethod(args, is_regression=True)
model.fit(data, info, train=True)

# Predict
vres, metric_name, predictions = model.predict(test_data, info, 'best-val')

# BUG: Predictions are in normalized space!
print(f"Predictions: mean={predictions.mean():.2f}, std={predictions.std():.2f}")
# Expected: mean≈50, std≈20
# Actual: mean≈0, std≈1  ⚠️

# To get correct predictions, user must manually denormalize:
predictions_correct = predictions * std + mean
```

## Expected vs Actual Behavior

**Expected:**
```python
predictions = model.predict(X_test)
# predictions in original scale: [45.2, 53.1, 48.9, ...]
```

**Actual:**
```python
predictions = model.predict(X_test)  
# predictions in normalized space: [-0.24, 0.16, -0.05, ...]
# ⚠️ Users must manually denormalize!
```

## Evidence from Real Dataset (HELOC, LGD task)

Target: Loss Given Default in [0, 1], mean=0.679, std=0.430

| Statistic | Ground Truth | Raw Predictions | After Manual Denorm |
|-----------|-------------|-----------------|-------------------|
| Mean      | 0.679       | **-0.001** ⚠️    | 0.680 ✓          |
| Std       | 0.430       | 0.622          | 0.263            |
| Range     | [0, 1]      | **[-1.96, 0.77]** ⚠️ | [-0.16, 1.01] ✓ |
| R²        | -           | -2.79 ⚠️         | 0.39 ✓           |

Without denormalization:
- Predictions centered at 0 instead of 0.679
- 50% of predictions are negative (impossible for LGD)
- R² = -2.79 (catastrophically wrong)

After manual denormalization:
- R² improves to 0.39
- Predictions in correct range

### Option 1: Fix in predict() methods (RECOMMENDED)
```python
def predict(self, data, info, model_name):
    N, C, y = data
    self.data_format(False, N, C, y)
    test_label = self.y_test
    test_logit = self.model.predict(self.N_test)
    
    # ADD DENORMALIZATION
    if self.is_regression and self.y_info['policy'] == 'mean_std':
        test_logit = test_logit * self.y_info['std'] + self.y_info['mean']
    
    vres, metric_name = self.metric(test_logit, test_label, self.y_info)
    return vres, metric_name, test_logit
```

**Apply this fix to:**
- `TALENT/model/classical_methods/lr.py` (LinearRegressionMethod)
- `TALENT/model/classical_methods/knn.py` (KnnMethod)  
- `TALENT/model/classical_methods/xgboost.py` (XGBoostMethod)
- `TALENT/model/classical_methods/svm.py` (SvmMethod)
- Any other classical regression methods

### Option 2: Fix in metric() method (ALTERNATIVE)

Denormalize predictions before computing metrics:
```python
def metric(self, predictions, labels, y_info):
    # Denormalize predictions if needed
    if self.is_regression and y_info['policy'] == 'mean_std':
        predictions = predictions * y_info['std'] + y_info['mean']
        labels = labels * y_info['std'] + y_info['mean']
    
    # Now compute metrics on original scale
    mae = skm.mean_absolute_error(labels, predictions)
    rmse = skm.mean_squared_error(labels, predictions) ** 0.5
    r2 = skm.r2_score(labels, predictions)
    
    return (mae, r2, rmse), ("MAE", "R2", "RMSE")
```

## Backward Compatibility

This is a **breaking change** but necessary for correctness. Users who have implemented manual denormalization will get duplicate denormalization. 

**Migration guide:**
1. Remove manual denormalization code: `y_pred_original = y_pred * std + mean`
2. Use predictions directly from TALENT
3. Update documentation to reflect this fix

## Related Issues
- May also affect deep learning regression methods - needs audit
- Same issue might exist in other preprocessing pipelines (quantile normalization, etc.)

## Environment
- TALENT version: [latest from main branch]
- Python: 3.10
- Discovered during: Credit risk modeling research (LGD regression)

## Additional Notes
This bug is difficult to detect because:
1. R² is scale-invariant (same in normalized or original space)
2. MAE/RMSE are scaled back in metric(), giving correct error magnitudes
3. Only becomes obvious with bounded targets requiring post-processing clipping
4. Deep learning methods may handle this differently, masking the issue

Thank you for this excellent toolkit! We use it extensively in our research and discovered this issue while benchmarking foundation models for credit risk prediction.
